Dapper Blood Oyster

High

# `CrossChainRouter`.`_handleLiquidationExecute` missed to convert `destlToken` for destination chain when using `_send`

### Summary

`_handleLiquidationExecute`,`_sendLiquidationFailure` used `destlToken` for `_send` but it is not available in destination chain. It is available for only current chain.
It must be converted to `lToken` for destination chain.

### Root Cause

In `CrossChainRouter`, for liquidation, `liquidator` calls `liquidateCrossChain` and it sends message to destination in [`_executeLiquidationCore`](https://github.com/sherlock-audit/2025-05-lend-audit-contest/blob/main/Lend-V2/src/LayerZero/CrossChainRouter.sol#L264-L285) function
```solidity
_send(
	params.srcEid,
	seizeTokens,
	params.storedBorrowIndex,
	0,
	params.borrower,
@>	lendStorage.crossChainLTokenMap(params.lTokenToSeize, params.srcEid), // Convert to Chain A version before sending
	msg.sender,
	params.borrowedAsset,
	ContractType.CrossChainLiquidationExecute
);
```
`lTokenToSeize` converted to `destlToken` for destination chain.
https://github.com/sherlock-audit/2025-05-lend-audit-contest/blob/main/Lend-V2/src/LayerZero/LendStorage.sol#L80
In destination chain, [`_lzReceive`](https://github.com/sherlock-audit/2025-05-lend-audit-contest/blob/main/Lend-V2/src/LayerZero/CrossChainRouter.sol#L772-L777) processes message using `_handleLiquidationExecute`, `_sendLiquidationFailure`.
[`_handleLiquidationExecute`](https://github.com/sherlock-audit/2025-05-lend-audit-contest/blob/main/Lend-V2/src/LayerZero/CrossChainRouter.sol#L355-L365), [`_sendLiquidationFailure`](https://github.com/sherlock-audit/2025-05-lend-audit-contest/blob/main/Lend-V2/src/LayerZero/CrossChainRouter.sol#L492-L502) sends message to source chain but they didn't convert token for other chain.
```solidity
function _handleLiquidationExecute(LZPayload memory payload, uint32 srcEid) private {
	...
	_send(
	srcEid,
	payload.amount,
	0,
	0,
	payload.sender,
@>	payload.destlToken, // @audit not converted for other chain
	payload.liquidator,
	payload.srcToken,
	ContractType.LiquidationSuccess
	);
}
function _sendLiquidationFailure(LZPayload memory payload, uint32 srcEid) private {
	_send(
		srcEid,
		payload.amount,
		0,
		0,
		payload.sender,
@>		payload.destlToken, // @audit not converted for other chain
		payload.liquidator,
		payload.srcToken,
		ContractType.LiquidationFailure
	);
}
```
This causes reverts for other chain because `lToken` is not exist for other chain.
Liquidation always fail.
The message is returned to source chain again and it processed in `_handleLiquidationSuccess`, `_handleLiquidationFailure`.
```solidity
function _handleLiquidationSuccess(LZPayload memory payload) private {
	// Find the borrow position on Chain B to get the correct srcEid
@>	address underlying = lendStorage.lTokenToUnderlying(payload.destlToken); 
	// @audit underlying is 0 because payload.destlToken is not exist in source chain.
	...
}
function _handleLiquidationFailure(LZPayload memory payload) private {
@>	address underlying = lendStorage.lTokenToUnderlying(payload.destlToken);
	...
}
```

### Internal Pre-conditions

.

### External Pre-conditions

.

### Attack Path

.

### Impact

`liquidateCrossChain` always fails. And also it processed in only destination chain. It causes confusion.

### PoC

_No response_

### Mitigation

Replace `payload.destlToken` to `lendStorage.crossChainLTokenMap(params.lTokenToSeize, srcEid)`